script.kodi.android.update/service.py

```diff
@@ -38,37 +39,50 @@ def log(msg, level=xbmc.LOGDEBUG):
 
 class Service(object):
     def __init__(self):
+        self.setSettings()
+        lastPath = REAL_SETTINGS.getSetting("LastPath") # CACHE = Keep last download, CLEAN = Remove all downloads
+        if not CACHE and CLEAN and xbmcvfs.exists(lastPath): self.deleteLast(lastPath)
+               
+               
+    def deleteLast(self, lastPath):
+        log('deleteLast')
         try:
-            self.getBuild()
-            self.getVersion()
-            self.lastPath = REAL_SETTINGS.getSetting("LastPath") # CACHE = Keep last download, CLEAN = Remove all downloads
-            if not CACHE and CLEAN and xbmcvfs.exists(self.lastPath):
-                log('Service, removing last download')
-                xbmcvfs.delete(self.lastPath)
-                xbmcgui.Dialog().notification(ADDON_NAME, LANGUAGE(30007), ICON, 4000)
-        except Exception as e: log("Service Failed! " + str(e), xbmc.LOGERROR)
+            xbmcvfs.delete(self.lastPath)
+            xbmcgui.Dialog().notification(ADDON_NAME, LANGUAGE(30007), ICON, 4000)
+        except Exception as e: log("deleteLast Failed! " + str(e), xbmc.LOGERROR)
+            
+            
+    def setSettings(self):
+        log('setSettings')
+        [func() for func in [self.getBuild,self.getPlatform,self.getVersion]]
         
         
-    def getBuild(self): 
+    def getBuild(self):
+        log('getBuild')
+        REAL_SETTINGS.setSetting("Build",json.dumps(json.loads(xbmc.executeJSONRPC(VER_QUERY) or '').get('result',{}).get('version',{})))
+        
+            
+    def getPlatform(self): 
+        log('getPlatform')
         count = 0
-        build = ''
         try:
-            while not xbmc.Monitor().abortRequested() and count < 3:
-                count += 1
+            while not xbmc.Monitor().abortRequested() and count < 15:
```

Comment:
maybe create and reuse the same `xbmc.Monitor()` object

Reply:
I agree... lazy functions to get the job done. I'll include a single call in the next iteration.